web/models/terms.ex

```diff
@@ -12,17 +12,27 @@ defmodule CoursePlanner.Terms do
   end
 
   def new do
-    Term.changeset(%Term{})
+    Term.changeset(%Term{holidays: [], courses: []})
   end
 
   def create_term(params) do
     %Term{}
     |> Term.changeset(params)
+    |> Changeset.put_assoc(:offered_courses, course_changesets(params))
     |> Repo.insert
   end
 
+  def course_changesets(%{"course_ids" => ids}) do
+    Enum.map(ids, fn id ->
+      Changeset.cast(%OfferedCourse{}, %{course_id: id}, [:course_id])
```

Comment:
I gave some thought to it, maybe this is not even the right association to use here, probably it is better to fetch the courses and use the `has_many :courses` association.

I will give it a try